exec/java-exec/src/main/java/org/apache/drill/exec/store/security/CredentialProviderUtils.java

```diff
@@ -43,4 +43,59 @@ public static CredentialsProvider getCredentialsProvider(
     }
     return new PlainCredentialsProvider(mapBuilder.build());
   }
+
+  /**
+   * Constructor for OAuth based authentication.  Allows for tokens to be stored in whatever vault
```

Comment:
I think we can avoid having all of this OAuth-specific code show up here like this, by a universally-supported `setCredential()`